### PR TITLE
Systems manager jh2

### PR DESCRIPTION
**Pull request by jhseceng**
Resolved conflict. Please review

### CODE REVIEW

File: systems-manager/Packaging-utilities/examples/linux-sensor-binary/CS_AMAZON/install.sh [hunk 1]
```diff
@@ -1,66 +1,172 @@
...
-#Get configs params
-echo 'Getting required config params ...'
-
-if [[ -z "$SSM_CS_CCID" ]]; then
-  echo "Missing required param SSM_CS_CCID. Ensure the target instance is running the latest SSM agent version"
-  exit 1
-fi
-CCID="${SSM_CS_CCID}"
+#!/bin/bash
+filename="falcon-sensor-6.16.0-11308.amzn2.x86_64.rpm"
```
Comment (shawndwells) [inline systems-manager/Packaging-utilities/examples/linux-sensor-binary/CS_AMAZON/install.sh:2]: Hard coded filename

File: systems-manager/Packaging-utilities/examples/linux-sensor-binary/packager.py [hunk 2]
```diff
@@ -278,6 +279,10 @@ def _get_digest(self, files):
...
             "Skipping AWS upload: please provide --aws_region, --ssm_automation_doc_name, and --s3bucket command-line options for upload")
         print("Package has been built successfully.")
-    else:
+    elif region and s3bucket:
+        S3BucketUpdater(region).update(s3bucket, files)
+    elif region and s3bucket and package_name:
         S3BucketUpdater(region).update(s3bucket, files)
         SSMPackageUpdater(region).update(package_name, "./s3-bucket/manifest.json")
+    else:
+        print("Nothing to do ... specify region + bucket or region + bucket + package_name")
```
Comment (shawndwells) [inline systems-manager/Packaging-utilities/examples/linux-sensor-binary/packager.py:288]: How are these options received? Can an example be given?

eg ``specify a region with the --region <region> flag``

File: systems-manager/Packaging-utilities/examples/linux-sensor-download/CS_AMAZON/install.sh [hunk 3]
```diff
@@ -78,7 +82,7 @@ aptInstall(){
    apt-get -y install libnl-genl-3-200 libnl-3-200
    sudo dpkg -i $1
    apt-get -y --fix-broken install
-   /opt/CrowdStrike/falconctl -s -f --cid=$2
+   /opt/CrowdStrike/falconctl -s -f --cid=$2 $3 $4
    service falcon-sensor restart
```
Comment (shawndwells) [inline systems-manager/Packaging-utilities/examples/linux-sensor-download/CS_AMAZON/install.sh:86]: Can more identifiable variable names be used?

File: systems-manager/Packaging-utilities/examples/linux-sensor-download/CS_AMAZON/install.sh [hunk 4]
```diff
@@ -172,6 +191,10 @@ case "$OS_NAME" in
 esac
 
 OS_VERSION=2
+if [[ "$(uname -p)" != *86* ]]
+then
+    OS_VERSION="$OS_VERSION - arm64"
+fi
```
Comment (shawndwells) [inline systems-manager/Packaging-utilities/examples/linux-sensor-download/CS_AMAZON/install.sh:197]: This logic will break for the pending s390x release

File: systems-manager/documentation/AWS-Systems-Manager-Intro.md [hunk 5]
```diff
@@ -246,6 +246,30 @@ The example contains the directories CS_AMAZON and CS_WINDOWS. Each directory sh
 * Uninstall script
 * Sensor Binary for the particular OS
 
+The Install Script At the top of the install script enter values for the the following
+filename=<<filename of the rpm package>>
+TARGET_VER=<<minimum version>> example format =5.38.10404
```
Comment (shawndwells) [inline systems-manager/documentation/AWS-Systems-Manager-Intro.md:251]:
```suggestion
Update the following variables at the top of the installation script:

```
filename=<<filename of the rpm package>>
TARGET_VER=<<minimum version>> example format =5.38.10404
```
```

File: systems-manager/documentation/AWS-Systems-Manager-Intro.md [hunk 6]
```diff
@@ -246,6 +246,30 @@ The example contains the directories CS_AMAZON and CS_WINDOWS. Each directory sh
...
+  FALCON_VER=$(sudo /opt/CrowdStrike/falconctl -g --version | awk -F= '{ print  $2}' | awk -F. '{ print  $1$2$3$4}')
+  if [ "$FALCON_VER" -lt "$TARGET_VER" ]; then
+    echo "Install required"
+    install
+  else
+    echo "Install not required"
+    exit
+  fi
+else
+  echo "Falcon-sensor not found"
```
Comment (shawndwells) [inline systems-manager/documentation/AWS-Systems-Manager-Intro.md:267]:
```suggestion
  echo "Package falcon-sensor not found"
```

**Review by shawndwells — CHANGES_REQUESTED** [review]
* Should systems-manager/Packaging-utilities/examples/linux-sensor-binary/CS_AMAZON/install.sh be dropped? Worried we keep re-creating bespoke bash installation scripts instead of using the centralized ones

**Comment by shawndwells** [comment]
Also looks like there are several merges included. You may want to rebase your branch and squash your incremental commits.

**Comment by jhseceng** [comment]
Creating new pull request after further updates

**Pull request closed without merging**